ComPact.ConsumerTests/MessagePactTests.cs

```diff
@@ -78,6 +78,43 @@ public async Task ShouldCreateMessagePactAfterVerifyingConsumer()
             File.Delete(pactDir + "messageConsumer-messageProvider.json");
         }
 
+        [TestMethod]
+        public async Task ShouldCreateMessagePactAfterVerifyingConsumerAsynchronously()
+        {
+            var handler = new RecipeAddedHandler();
+
+            var fakePactBrokerMessageHandler = new FakePactBrokerMessageHandler();
+            fakePactBrokerMessageHandler.Configure(HttpMethod.Put, "http://localhost:9292").RespondsWith(HttpStatusCode.Created);
+            var publisher = new PactPublisher(new HttpClient(fakePactBrokerMessageHandler) { BaseAddress = new Uri("http://localhost:9292") }, "1.0", "local");
+
+            var builder = new MessagePactBuilder("messageConsumer", "messageProvider", publisher);
+            var messsageBuilder = await _messageBuilder
+                .VerifyConsumerAsync<RecipeAdded>(m => handler.HandleAsync(m));
+
+            await builder.SetUp(messsageBuilder)
+                .BuildAsync();
+
+            // Check if handler has been called
+            Assert.IsNotNull(handler.ReceivedRecipes.FirstOrDefault());
+            Assert.AreEqual("A Recipe", handler.ReceivedRecipes.First().Name);
+
+            // Check if pact has been published and tagged
```

Comment:
I think this can be left out of the test. It's already being tested in another scenario so it's a bit redundant.

Reply:
fixed